src/MQTTAsyncUtils.c

```diff
@@ -2058,23 +2084,31 @@ thread_return_type WINAPI MQTTAsync_receiveThread(void* n)
 		pack = MQTTAsync_cycle(&sock, timeout, &rc);
 		MQTTAsync_lock_mutex(mqttasync_mutex);
 		if (MQTTAsync_tostop)
+		{
+			MQTTAsync_freePack(pack);
 			break;
+		}
 		timeout = 1000L;
 
 		if (sock == 0)
+		{
+			MQTTAsync_freePack(pack);
 			continue;
+		}
 		/* find client corresponding to socket */
 		if (ListFindItem(MQTTAsync_handles, &sock, clientSockCompare) == NULL)
 		{
 			Log(TRACE_MINIMUM, -1, "Could not find client corresponding to socket %d", sock);
 			/* Socket_close(sock); - removing socket in this case is not necessary (Bug 442400) */
+			MQTTAsync_freePack(pack);
 			continue;
 		}
 		m = (MQTTAsyncs*)(MQTTAsync_handles->current->content);
 		if (m == NULL)
 		{
 			Log(LOG_ERROR, -1, "Client structure was NULL for socket %d - removing socket", sock);
 			Socket_close(sock);
+			MQTTAsync_freePack(pack);
```

Comment:
_pack_ is going out of scope without transfer. (pack may or may not point to memory)